common/src/api/external/mod.rs

```diff
@@ -2551,6 +2551,74 @@ pub struct SwitchPortConfig {
     pub geometry: SwitchPortGeometry,
 }
 
+/// The speed of a link.
+#[derive(Copy, Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum LinkSpeed {
```

Comment:
This is not net-new, it's moved from `nexus/external_api/params.rs` to be used as a common data structure for PUT/POST and GET requests.